### ISSUE TITLE
[tt-train] MuonWithAdamW.get_lr / set_lr only address the Muon LR

### ISSUE THREAD

**Issue by awliu-TT** [tenstorrent/tt-metal#56027]
## Summary

`MuonWithAdamW` (`tt-train/sources/ttml/ttml/common/muon_optimizer.py`) wraps two inner
optimizers with two independent learning rates: Muon for 2D hidden-layer weights and
AdamW for everything else (embeddings, norms, biases, LM head). Its `get_lr()`/`set_lr()`
delegate **only to the inner Muon optimizer**:

```python
def get_lr(self):
    return self._muon.get_lr()

def set_lr(self, lr):
    self._muon.set_lr(lr)
```

Nothing stops generic LR-scheduling code from treating it as a single-LR optimizer, and
two code paths do exactly that today:

1. **Python LR schedulers** (`ttml/common/schedulers.py`): `_SchedulerBase.__init__`
   records `optimizer.get_lr()` as the base LR and every `step()` writes via
   `optimizer.set_lr(...)`. Attaching e.g. `CosineAnnealingScheduler` to a
   `MuonWithAdamW` instance works without error but schedules only the Muon LR.
2. **Trainer schedules**: `SFTTrainer` applies its `lr_schedule` callable by calling
   `self._optimizer.set_lr(lr)` every step
   (`tt-train/sources/ttml/ttml/trainers/sft_trainer.py`), and
   `train.py::build_lr_schedule` (`tt-train/sources/examples/train/train.py`) shapes the
   curve from `optimizer.get_lr()` — i.e. from the Muon LR only. `grpo_trainer`'s manual
   warmup calls `set_lr` directly as well.

In every case the AdamW LR is silently never scheduled. There is also no supported way to
schedule it: the inner optimizers (`_muon`, `_adamw`) are private, and the only public
knob is a manual `set_adamw_lr()` that no scheduler or trainer knows about.

In the currently-shipped `SFTTrainer` (no `scheduler_type`), the config is benign only by accident: the schedule is constant, so `set_lr` rewrites the same Muon LR each step.

## Design question

What contract should `get_lr`/`set_lr` honor on a fused two-LR optimizer? Candidates:

1. **Reject scheduling on the composite optimizer wrapper**: raise from `set_lr` (or add a capability flag
   that LR schedulers/trainers check), and expose the inner optimizers publicly so callers
   attach one scheduler to each — mirroring PyTorch core's idiom of a separate
   `torch.optim.Muon` + `torch.optim.AdamW`, each with its own scheduler. Fails loudly,
   but breaks the wrapper's substitutability as a plain `OptimizerBase`.
2. **Proportional scaling**: `set_lr(lr)` scales *both* inner LRs by `lr / base_lr` from
   their recorded bases; `get_lr()` returns the Muon (reference) LR. This is a hand-rolled
   version of what PyTorch's per-param-group schedulers do automatically (cf. the
   reference `MuonWithAuxAdam` in KellerJordan/Muon, which is scheduler-safe precisely
   because each param group carries its own lr/base). Keeps existing trainer code working,
   but needs careful base-LR bookkeeping across checkpoint save/restore.
3. **Param-group-style scheduling** in `OptimizerBase` and the schedulers (list-of-LRs
   contract like `torch.optim.lr_scheduler`). Most general, largest change.

Whatever the choice, the trainer paths (`SFTTrainer`, `grpo_trainer`) should either work
correctly with this optimizer or fail loudly — not silently schedule half the parameters.

## References

- `tt-train/sources/ttml/ttml/common/muon_optimizer.py` — wrapper; `get_lr`/`set_lr` delegate to Muon only
- `tt-train/sources/ttml/ttml/common/schedulers.py` — `_SchedulerBase` attaches to any optimizer via `get_lr`/`set_lr`
- `tt-train/sources/ttml/ttml/trainers/sft_trainer.py` — `set_lr` driven from `lr_schedule` every step
- `tt-train/sources/examples/train/train.py` — `build_lr_schedule` reads `get_lr()` only
- `tt-train/configs/training_configs/training_shakespeare_tinyllama_muon.yaml` — only config using `MuonWithAdamW`

**Comment by awliu-TT** [comment]
Option 1 taken as a patch for now in https://github.com/tenstorrent/tt-metal/pull/54320 to unblock. We can make changes if necessary.
